### PR TITLE
[bazel] Add capability to execute jtag/RAM tests 

### PR DESCRIPTION
**Pull request by cfrantz**
1. RAM-based binaries need only produce the `.elf` file.
2. Create a `cw310_jtag_params` helper that includes the necessary
   dependencies and flags for a JTAG test dispatch via opentitantool.
3. Convert a RAM/JTAG test to the new rules.

Note: this PR depends on #19650, #19697 and #19710.  You need only review the last two commits.

### CODE REVIEW

**Review by pamaury — APPROVED** [review]

**Review by timothytrippel — APPROVED** [review]

**Comment by timothytrippel** [comment]
> 1. RAM-based binaries need only produce the `.elf` file.

I guess we can eventually remove the option to load SRAM VMEM programs then: https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/test_utils/load_sram_program.rs;drc=05ba6772e04509a9ed133e692cda16d6dc9cd309;l=36
(once everything switches over)?

File: rules/opentitan/fpga_cw310.bzl [hunk 1]
```diff
@@ -2,14 +2,30 @@
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
 
-load("@lowrisc_opentitan//rules/opentitan:providers.bzl", "Cw310BinaryInfo")
+load(
+    "@lowrisc_opentitan//rules/opentitan:providers.bzl",
+    "Cw305BinaryInfo",
+    "Cw310BinaryInfo",
+    "Cw340BinaryInfo",
```
Comment (moidx) [inline rules/opentitan/fpga_cw310.bzl:9]: nit: This is in the direction of renaming this file as `fpga_cw` or `fpga_chipwhisperer`.

File: rules/opentitan/openocd.bzl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "//util/openocd/target:lowrisc-earlgrey-lc.cfg",
+]
+
+OPENTITANTOOL_OPENOCD_TEST_CMD = """
+    --openocd="$(rootpath //third_party/openocd:openocd_bin)"
+    --openocd-adapter-config="$(rootpath //third_party/openocd:jtag_olimex_cfg)"
+    --openocd-riscv-target-config="$(rootpath //util/openocd/target:lowrisc-earlgrey.cfg)"
+    --openocd-lc-target-config="$(rootpath //util/openocd/target:lowrisc-earlgrey-lc.cfg)"
+    --clear-bitstream
+    --bitstream={bitstream}
```
Comment (moidx) [inline rules/opentitan/openocd.bzl:18]: This implies that this config is used for FPGA targets. Should we capture that in the variable name?

I think also from a SiVal perspective, would we use a separate TEST_CMD? In which case it is probably ok to make this one fpga specific.

**Review by moidx — APPROVED** [review]

File: rules/opentitan/openocd.bzl (hunk 2 above)
Comment (cfrantz) [reply rules/opentitan/openocd.bzl:18]: Currently, FPGA targets are the only use-case.  I agree that we should refactor this a bit to work for both FPGA and SiVal targets.

File: rules/opentitan/fpga_cw310.bzl (hunk 1 above)
Comment (cfrantz) [reply rules/opentitan/fpga_cw310.bzl:9]: Yep. #19900 suggests doing this rename.

File: rules/opentitan/openocd.bzl (hunk 2 above)
Comment (cfrantz) [reply rules/opentitan/openocd.bzl:18]: Added a note to #19906.

**Pull request merged**